### PR TITLE
:mortar_board: Topic14 Aside on inheriting problems 

### PR DESCRIPTION
**Pull request by jameshughes89**
### Related Issues or PRs
Related #138 

### What
Add an aside for inheriting from concrete classes and getting too excited to extend everything. This is nearly a copy/paste from @twentylemon's explination.

### Additional Notes
It's defo a longer PR than I had expected.

### CODE REVIEW

File: src/site/topic14-careful.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+* This *seems* great
+* We inherit the ``getLength``, ``getWidth``, and ``getArea``
+* We override the ``toString`` for our special needs
+* We even can even hijack the superclass' constructors with ``super()`` in a similar way to using ``this()``
+
+
+Liskov's Substitution Principle
+===============================
+
+* `"Functions that use pointers or references to base classes must be able to use objects of derived classes without knowing it" <https://en.wikipedia.org/wiki/Liskov_substitution_principle>`_
```
Comment (twentylemon) [inline src/site/topic14-careful.rst:119]: If you're mentioning this, you also definitely want to mention [SOLID](https://en.wikipedia.org/wiki/SOLID). Think just a sub-point here is fine, saying it's the L from SOLID design principals with another link.

File: src/site/topic14-careful.rst [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        } else {
+            rect.setWidth(2.0 * rect.getWidth());
+        }
+    }
+
+* Now this *solves* it
+* Except, `Hyrum's Law <https://www.hyrumslaw.com/>`_ says that all observable behaviours, intentional or not, will be depended on by somebody
+* So, someone out there depends on the fact that ``pumpItUp`` is quadrupling the ``Square``, even though it honestly shouldn't be
+
+* Ok, so, we can fix this by adding another method and changing ``pumpItBack`` for the person depending on the problematic functionality
```
Comment (twentylemon) [inline src/site/topic14-careful.rst:186]: `pumpItBack` -> `myPumpItUp` ?

File: src/site/topic14-careful.rst [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+* NOW this *solves* it
+* But, now we have a function saying: if it's a ``Square`` do one thing, if it's a ``Rectangle`` do another thing
+    * So... it would seem that for our needs here, a ``Square`` is **not** a ``Rectangle``
+* We also have two pieces of code trying to do the same thing
+    * What happens if ``Rectangle`` gets extended again? Write another version of the method?
+
+* We ended up doing a lot of extra work for no reason at all
+* Our code got more complex
+* It's going to be a lot easier to just not use inheritance here
```
Comment (twentylemon) [inline src/site/topic14-careful.rst:211]: It's probably also worth mentioning a better option: [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance). If you want to code share, a `Square` could _use_ a `Rectangle` and delegate the square specific logic to that.

File: src/site/topic14-careful.rst (hunk 2 above)
Comment (jameshughes89) [reply src/site/topic14-careful.rst:186]: how tf?

**Review by twentylemon — APPROVED** [review]

**Pull request merged**